### ISSUE TITLE
CTE Materialization hints are not properly parsed in SQLite dialect

### ISSUE THREAD

**Issue by karimudev** [sqlfluff/sqlfluff#8125]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLite has support for non-standard SQL syntax `AS MATERIALIZED` and `AS NOT MATERIALIZED` in CTE definitions. (see: [The WITH Clause: Materialization Hints](https://sqlite.org/lang_with.html#materialization_hints)).
This syntax originates from Postgres. 

SQLFluff parses these materialization hints properly for the Postgres dialect (see: [dialect_postgres.py:6058](https://github.com/sqlfluff/sqlfluff/blob/4.2.2/src/sqlfluff/dialects/dialect_postgres.py#L6058)) but not for SQLite.

### Expected Behaviour

The example should be properly linted for SQLite dialect.

### Observed Behaviour

With the example from the section below, SQLFluff returns the following:
```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'WITH cte
                       | AS MATERIALIZED (\n    SELECT co...'
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
```

In comparison, SQLFluff returns the following with option `--dialect postgres`

```
All Finished 📜 🎉!
```

### How to reproduce

`query.sql`
```sql
WITH cte AS MATERIALIZED (
    SELECT col FROM table
)

SELECT col FROM cte
```

Command to run: `sqlfluff lint --dialect sqlite query.sql`

### Dialect

SQLite

### Version

sqlfluff, version 4.2.2
python, 3.14.6

### Configuration

No configuration needed with the command provided above (or just a bare configuration file that sets `dialect` to `sqlite`)

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
